### PR TITLE
[Docs] Polish the docs

### PR DESCRIPTION
**Pull request by martintmk**
### Details on the issue fix or feature implementation

Run the new sections through Chat GPT 4 model to polish them off.

Contributes to #1558 

### Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

**Comment by martintmk** [comment]
cc @peter-csala

**Comment by peter-csala** [comment]
Hey @martintmk thanks, I'll check it later today.

File: docs/strategies/retry.md [hunk 1]
```diff
@@ -359,16 +361,16 @@ await retry.ExecuteAsync(ct =>
...
+If `DoSomething` and `BeforeEachAttempt` are interdependent, group them or craft a simple wrapper to invoke them in the correct sequence.
 
-Lets suppose we have an `HttpClient` which issues a request and then we try to parse a large Json
+### 6 - Having a single strategy for multiple failures
 
-❌ DON'T
+Suppose we have an `HttpClient` that issues a request and then we try to parse a large JSON.
 
-Have a single policy to cover everything
+❌ **Don't**
```
Comment (peter-csala) [inline docs/strategies/retry.md:371]: Why did you change `DON'T` to `**Don't**`?

File: docs/strategies/retry.md [hunk 2]
```diff
@@ -391,13 +393,12 @@ await pipeline.ExecuteAsync(async (ct) =>
...
 **Reasoning**:
-- In the previous point it was suggested that  _if the `X` and `Y` coupled together then decorate them together_
-   - only if they are all part of the same failure domain
-   - in other words a pipeline should cover one failure domain
 
-✅ DO
+Previously, it was suggested that you should combine `X` and `Y` only if they are part of the same failure domain. In simpler terms, a pipeline should address only one type of failure.
 
-Define a strategy per failure domain
+✅ **Do**
```
Comment (peter-csala) [inline docs/strategies/retry.md:399]: Why did you change `DO` to `**Do**`?

File: docs/strategies/retry.md (hunk 1 above)
Comment (martintmk) [reply docs/strategies/retry.md:371]: Fixed.

File: docs/strategies/retry.md (hunk 2 above)
Comment (martintmk) [reply docs/strategies/retry.md:399]: Fixed.

**Review by peter-csala — APPROVED** [review]
